7a_temp_coop_munge/src/data_parsers/parse_Iowa_DNR_LimnoProfiles_2000_2020.R

```diff
@@ -0,0 +1,507 @@
+    dplyr::select(! c(lake, lakeid, date)) #holdover column from 2017-2020 data
+
+  # out -------------------
+  saveRDS(object = data_clean, file = outfile)
+  sc_indicate(ind_file = outind, data_file = outfile)
+}
+
+
+
+# Parsers by subfolder -----------------------------------------
+
+#' Parse Iowa DNR data from 2000-2007
+#'
+#' This data does not contain lake id, date, or time in the data file. Instead,
+#' these values are extracted from the name of the file. This parser also
+#' accounts for profiles that include data from the descent _and_ the ascent
+#' of the data logger. According to cooperators, the ascent data should
+#' be removed.
+#'
+#' @param x chr, full file path
+#' @param skip_rows int, number of rows to skip when reading in data. Be sure to . Start the
+#' @param keep_cols chr, vector of column names to keep.
+#' @param subset_lakeid num, vector for start/stop values passed to `base::substr` in `extract_lakeid`
+#' Column name spelling should match. Case does not matter because
+#' all values to be changed `tolower` within the function.
+#'
+parse_2000_2009_data <- function(x, skip_rows = 0, keep_cols,
+                                 subset_lakeid = c(6, 9)) {
+
+  dat <- suppressMessages(readxl::read_xls(x, skip = skip_rows, col_names = F))
+  names(dat) <- suppressMessages(readxl::read_xls(x, n_max = 1, col_names = F)) %>%
+    unlist %>%
+    .[c(1:ncol(dat))] %>%  # some data sets have an empty last row
+    tolower
+
+  dat <- dat %>%
+    dplyr::select(all_of(tolower(keep_cols))) %>%
+    dplyr::mutate(depth = as.numeric(depth), temp = as.numeric(temp))
+
+  # add LakeID and date columns from file name if necessary
+  dat$lakeid <- extract_lakeid(x, start = subset_lakeid[1],
+                               stop = subset_lakeid[2])
+
+  # wrangle date and time
+  df_names <- names(dat)
+
+  ## No date field present
+  if(!any(c('date', 'sampling_date_time') %in% df_names)) {
+    dat$date <- extract_date(x, subset_yr = c(1,4), subset_jday = c(15, 17))
+  }
+
+  if('time' %in% names(dat)) {
+    # note some profiles contain time in HMS and values are reported every 1 sec
+    # currently all data is left in (e.g. no aggregation), so you may see data
+    # at multiple time steps
+    dat$time <- format(dat$time, "%H:%M")
+  }
+
+  # datetime present
+  if('sampling_date' %in% df_names){
+    dat$date <- as.Date(dat$sampling_date_time)
+    dat$time <- format(dat$sampling_date_time, "%H:%M")
+  }
+
+  # Check to see if data.frame has data
+  dat_rows <- nrow(dat)
+
+  if(dat_rows > 1) {
+    # check max depth and trim data if necessary
+    if(check_final_depth(dat$depth)) {
+      depth_loc <- which(names(dat) == 'depth') # id position of the depth column
+      dat <- trim_to_max_depth(dat, depth_col = depth_loc)
+
+      # return(dat)
+    }
+  } else {
+    print(paste('File ', basename(x),
+                ' appears to be empty or has problematic data. Data only has ',
```

Comment:
I had 
```r
[1] "File 2005_L113_2005173014.xls appears to be empty or has problematic data. Data only has 1 row(s) of data" 
```
I prefer `message` to `print` in these situations, perhaps switch to that?

Reply:
Resolved